demos/src/Experiments/Figure/Vue/figure.ts

```diff
@@ -177,6 +181,48 @@ export const Figure = Node.create<FigureOptions>({
       },
     }
   },
+  
+  addKeyboardShortcuts() {
+    return {
+      Enter: ({ editor }) => {
+        if (!this.options.exitOnEnter) {
+          return false
+        }
+
+        const { selection } = editor.state;
+        const { $from, empty } = selection;
+
+        if (!empty || $from.parent.type !== this.type) {
+          return false;
+        }
+
+        return editor.commands.command(({ state, dispatch }) => {
+          const { $head, $anchor } = state.selection
+
+          if (!$head.sameParent($anchor)) {
+            return false
+          }
+
+          const above = $head.node(-1)
+          const after = $head.indexAfter(-1)
+          const type = defaultBlockAt(above.contentMatchAt(after))
+          
+          if (!type || !above.canReplaceWith(after, after, type)) { 
+            return false
+          }
+          
+          if (dispatch) {
+            const pos = $head.after()
+            const tr = state.tr.replaceWith(pos, pos, type.createAndFill()!)
+
+            tr.setSelection(Selection.near(tr.doc.resolve(pos), 1))
+            dispatch(tr.scrollIntoView())
+          }
+          return true
+        })
```

Comment:
This could be its own command "exitBlockNode". The same thing is used in the `code-block` extension.